app/ui-react/syndesis/cypress/integration/create-integration.spec.js

```diff
@@ -0,0 +1,281 @@
+describe('Create an Integration', () => {
+  const randomInteger = () => {
+    return Math.floor(Math.random() * (100 + 1));
+  };
+  const nameInt = randomInteger();
+  const connectionName = 'E2E Todo Connection ' + nameInt;
+  const integrationName = 'E2E Todo Integration ' + nameInt;
+  const connectionSlug = 'e2e-todo-connection-' + nameInt;
+  const integrationSlug = 'e2e-todo-integration-' + nameInt;
+
+  function createConnection() {
+    /**
+     * Create To Do Connection
```

Comment:
I kinda think we're gonna find this is too much setup for this test.  Another option might be to upload an exported integration and delete the integration, which would then leave the connections in-place to be used by the integration test.  The problem here is that this basically duplicates the create connection test, but if that test fails, this test will also fail, but not because creating an integration is broken, but instead because creating a connection is broken.

Reply:
I've been thinking about this a lot. I completely agree with you about the setup for the test, it's too much. The problem was that we wanted to keep each E2E test completely self-contained, but we didn't want to use mock data. We wanted to test the actual API though, so it's a bit of a catch 22.

With tapes, we can probably get rid of this function, so long as tests are _always_ run with the tape (which will guarantee that we have a certain connection/integration/whatever there), but there is no direct interaction with the API when you use the tapes, defeating part of the purpose of the test. Could you point me to some more info about how the `--test-support` flag works and if that might be a good alternative for working with data? Otherwise I'm leaning more toward swagger-generated mock data tbh. cc @riccardo-forina 

This is gonna be a problem for integration tests as well, so the sooner we come up with something, the better.

Reply:
btw I'm continuing to write tests because despite which route we take, assertions should stay the same and tests can still be run (just maybe not ready for CI/CD or git hooks).

Reply:
oh of course!  Yeah that makes sense.  We can ignore the noise I think, as some tests are better than no tests :smile: